src/main/java/com/mraof/minestuck/skaianet/SessionHandler.java

```diff
@@ -40,120 +70,227 @@ public abstract class SessionHandler
+	void newPredefineData(PlayerIdentifier player)
+	{}
+	
+	Map<Integer, String> getServerList(PlayerIdentifier client)
 	{
-		return session.completed;
+		Optional<PlayerIdentifier> primaryServer = skaianetData.connections.primaryPartnerForClient(client);
+		Map<Integer, String> map = new HashMap<>();
+		for(PlayerIdentifier server : skaianetData.computerInteractions.openServerPlayers())
+		{
+			
+			if(primaryServer.isPresent() && primaryServer.get().equals(server)
+					|| primaryServer.isEmpty() && skaianetData.connections.canMakeNewRegularConnectionAsServer(server)
+					|| skaianetData.connections.canMakeSecondaryConnection(client, server))
+				map.put(server.getId(), server.getUsername());
+		}
+		return map;
 	}
 	
-	void onConnectionChainBroken(Session session)
-	{}
-	
 	/**
-	 * Will check if two players can connect based on their main connections and sessions.
-	 * Does NOT include session size checking.
-	 * @return True if client connection is not null and client and server session is the same or 
-	 * client connection is null and server connection is null.
+	 * @return all players whose title or land types (predefined or not) should influence the selection of the same for the specified player.
 	 */
-	//TODO repalce with proper checks when creating a regular or secondary connection
-	private boolean canConnect(PlayerIdentifier client, PlayerIdentifier server)
+	Stream<PlayerIdentifier> playersToCheckForDataSelection(PlayerIdentifier player)
+	{
+		Session session = this.getPlayerSession(player);
+		return session != null
+				? session.getPlayers().stream().filter(otherPlayer -> !otherPlayer.equals(player))
+				: Stream.empty();
+	}
+	
+	public static SessionHandler get(MinecraftServer server)
 	{
-		Session sClient = getPlayerSession(client), sServer = getPlayerSession(server);
-		Optional<SburbConnection> cClient = skaianetHandler.getPrimaryConnection(client, true);
-		Optional<SburbConnection> cServer = skaianetHandler.getPrimaryConnection(server, false);
-		boolean serverActive = cServer.isPresent();
-		if(!serverActive && sServer != null)
-			for(SburbConnection c : sServer.connections)
-				if(c.getServerIdentifier().equals(server))
-				{
-					serverActive = true;
-					break;
-				}
-		
-		return cClient.isPresent() && sClient == sServer && (MinestuckConfig.SERVER.allowSecondaryConnections.get() || cClient.get() == cServer.orElse(null))	//Reconnect within session
-				|| !cClient.isPresent() && !serverActive && !(sClient != null && sClient.locked) && !(sServer != null && sServer.locked);	//Connect with a new player and potentially create a main connection
+		return SkaianetData.get(server).sessionHandler;
 	}
 	
-	Session getSessionForConnecting(PlayerIdentifier client, PlayerIdentifier server) throws MergeResult.SessionMergeException
+	/**
+	 * An implementation that only allows for a single, global session, which all players are assumed to be part of
+	 */
+	private static final class Global extends SessionHandler
```

Comment:
Since I havent messed around with it before, have you tested the global config with these changes?

Reply:
Tested a little bit just now, but nothing thorough. I might look into making a few game tests for `SessionHandler`.